client/core/trade.go

```diff
@@ -547,18 +547,19 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 	// Prepare the asset.Contracts.
 	swaps := new(asset.Swaps)
 	for _, match := range matches {
-		dbMatch, _, proof, _ := match.parts()
+		dbMatch, _, proof, auth := match.parts()
 		value := dbMatch.Quantity
 		if !match.trade.Sell {
 			value = calc.BaseToQuote(dbMatch.Rate, dbMatch.Quantity)
 		}
+		matchTime := encode.UnixTimeMilli(int64(auth.MatchStamp))
+		lockTime := matchTime.Add(time.Hour * 24).UTC().Unix()
 		if dbMatch.Side == order.Maker {
 			proof.Secret = encode.RandomBytes(32)
 			secretHash := sha256.Sum256(proof.Secret)
 			proof.SecretHash = secretHash[:]
+			lockTime = matchTime.Add(time.Hour * 48).UTC().Unix()
```

Comment:
Yikes good catch.  But I feel like we should have these two constant times somewhere in a common server-client package.

Reply:
I actually thought about making them DEX cfg variables.

Reply:
> no law that says these couldn't be 2 hr / 4 hr

They need to guarantee (probabilistically) `SwapConf` confirmations within the taker's locktime. And since `SwapConf` is a variable, we need to be generous.

Reply:
Regarding making these constants, one reason I had in mind for making them dex cfg variables is to enable testing refund scenarios in #432.

An alternative would be declaring as `var` instead of `const` so the test code can change the values to something much shorter for test purposes.

A couple other reasons to consider making the locktimes dex cfg variables:
- If we make them dex cfg variables, it'd be easier for clients to know what's expected in terms of swap locktimes (just check the dex cfg to know what locktime to set). Short of that, we'd need to add those values to the spec so all parties know what's expected.
- Since `swapConf` is a dex cfg variable, it'd also make sense to make the locktime similarly variable and we could add some validation code to ensure that the set locktimes are sane and generous.

Reply:
@itswisdomagain You make some good arguments for making lock times config settings.  Let's consider that for future work.